### PR TITLE
CORE-541: Statically Define Built-in Networks

### PR DESCRIPTION
**Pull request by EBGToo**
Acknowledges that supported networks *must be* built-in by defining BRCryptoNetworkCanonicalType as one of BTC, BCH, ETH, XRP, HBAR, ...  Exposes this type as `network.type` and uses it in Demo code by replacing string comparisons with Currency.codeAsBTC, etc.  Remove Currency.codeAs*.  Defines NetworkType.

Adds `BCryptoNetwork* cryptoNetworkInstallBuiltins(size_t *count)` to install and to return the builtin networks (in C code).  Removes Swift and Java implementations in System (Swift) and Blockchains (Java) that created currencies, units and blockchains.

Built in networks are defined in crypto/BRCryptoConfig.h

Move addressScheme and syncMode functions out of System and on to Network, implemented in C with Swift/Java interfaces in Network.

Draft: Pending Unit tests

### CODE REVIEW

File: Java/corenative/src/main/java/com/breadwallet/corenative/crypto/BRCryptoNetwork.java [hunk 1]
```diff
@@ -25,65 +25,29 @@
...
-        return new BRCryptoNetwork(
-                CryptoLibraryDirect.cryptoNetworkCreateAsBTC(
-                        uids,
-                        name,
-                        isMainnet ? BRCryptoBoolean.CRYPTO_TRUE : BRCryptoBoolean.CRYPTO_FALSE
-                )
-        );
+    public static BRCryptoNetwork findBuiltin (String uids) {
+        Pointer builtin = CryptoLibraryDirect.cryptoNetworkFindBuiltin (uids);
+        if (null == builtin) throw new RuntimeException("Missed builtin: " + uids);
```
Comment (michael-brd) [inline Java/corenative/src/main/java/com/breadwallet/corenative/crypto/BRCryptoNetwork.java]: Purely if you are interested; for these types of runtime checks I use the Preconditions checks (checkNotNull, checkState, etc.). These manifest as runtime exceptions.

File: Java/corecrypto/src/main/java/com/breadwallet/corecrypto/Network.java [hunk 2]
```diff
@@ -33,62 +39,24 @@
...
-                    core = optional.get();
-                } else {
-                    throw new IllegalArgumentException("Unsupported ETH network");
-                }
-
-                break;
-            default:
-                core = BRCryptoNetwork.createAsGen(uids, name, currency.getCoreBRCryptoCurrency(), isMainnet);
-                break;
+    static Network findBuiltin (String uids) {
```
Comment (michael-brd) [inline Java/corecrypto/src/main/java/com/breadwallet/corecrypto/Network.java]: Purely if you are interested; for functions that can return null, I use the Optional type.

File: Java/corecrypto/src/main/java/com/breadwallet/corecrypto/Network.java [hunk 3]
```diff
@@ -264,11 +239,60 @@ public boolean hasCurrency(com.breadwallet.crypto.Currency currency) {
...
     }
 
+    public void addCurrency(com.breadwallet.crypto.Currency currency,
+                            com.breadwallet.crypto.Unit baseUnit,
+                            com.breadwallet.crypto.Unit defaultUnit) {
+    }
+
+    public void addUnitFor(com.breadwallet.crypto.Currency currency, com.breadwallet.crypto.Unit unit) {
+    }
+
```
Comment (michael-brd) [inline Java/corecrypto/src/main/java/com/breadwallet/corecrypto/Network.java]: Are these TODOs?

File: Java/corenative/src/main/java/com/breadwallet/corenative/crypto/BRCryptoNetwork.java (hunk 1 above)
Comment (EBGToo) [reply Java/corenative/src/main/java/com/breadwallet/corenative/crypto/BRCryptoNetwork.java]: It is a bit harsh to fail on a `uids` lookup...  Changed to Optional.

File: Java/corecrypto/src/main/java/com/breadwallet/corecrypto/Network.java (hunk 2 above)
Comment (EBGToo) [reply Java/corecrypto/src/main/java/com/breadwallet/corecrypto/Network.java]: Updated.

File: Java/corecrypto/src/main/java/com/breadwallet/corecrypto/Network.java (hunk 3 above)
Comment (EBGToo) [reply Java/corecrypto/src/main/java/com/breadwallet/corecrypto/Network.java]: Over looked; still need to run Unit tests and Java Demo App...  Doing now.

**Pull request merged**